flocker/node/_docker.py

```diff
@@ -527,6 +568,34 @@ def _address_in_use(self, apierror):
             return None
         return AddressInUse(address=(ip, int(port)), apierror=apierror)
 
+    def _image_data(self, image, container, state):
+        cached_data = self._image_cached_data(image)
+        if cached_data is not None:
+            return cached_data
+        try:
+            image_data = self._client.inspect_image(image)
+            self._image_cache[image] = image_data
+        except APIError as e:
+            if e.response.status_code == NOT_FOUND:
+                # Image has been deleted, so just fill in some
+                # stub data so we can return *something*. This
+                # should happen only for stopped containers so
+                # some inaccuracy is acceptable.
+                Message.new(
+                    message_type="flocker:docker:image_not_found",
+                    container=container, running=state
+                ).write()
+                image_data = {u"Config": {u"Env": [], u"Cmd": []}}
+                self._image_cache[image] = image_data
+            else:
+                raise
+        return image_data
+
+    def _image_cached_data(self, image):
```

Comment:
I don't think this method adds much. If the test needs to poke at private details it can poke at the data structure.